src/backend/cuda/platform.cpp

```diff
@@ -332,10 +342,10 @@ DeviceManager::DeviceManager()
 
     sortDevices();
 
-    for(int i = 0; i < nDevices; i++) {
-        setActiveDevice(i, cuDevices[i].nativeId);
-        CUDA_CHECK(cudaStreamCreate(&streams[i]));
-    }
+    // Initialize all streams to 0.
+    // Streams will be created in setActiveDevice()
+    for(int i = 0; i < (int)MAX_DEVICES; i++)
+        streams[i] = (cudaStream_t)0;
```

Comment:
Good catch, but can you change this to the following:
```
for(int i = 0; i < nDevices; i++) {
    // first call to the following function will automatically
    // create a stream for that particular device
    setActiveDevice(i, cuDevices[i].nativeId);
}
```
so that `cudaStreamCreate` is not called upon during actual code execution.